### PR TITLE
[inference] Support wint4 groupwise with cutlass gemm

### PR DESCRIPTION
**Pull request by freeliuzc**
### PR types

New features
### PR changes

OPs
### Description

PCard-77383

Related PR: https://github.com/PaddlePaddle/Paddle/pull/60204

### CODE REVIEW

**Comment by wwbitejotunn** [comment]
LGTM for Wint8 gemm

File: paddle/phi/kernels/fusion/cutlass/cutlass_kernels/cutlass_heuristic.h [hunk 1]
```diff
@@ -187,14 +207,34 @@ static CutlassGemmConfig estimate_best_config_from_occupancies(
   }
 
   CutlassGemmConfig best_config;
-  if (m >= 256 &&
+
+  if (m >= 256 && sm == 86 &&
```
Comment (wwbitejotunn) [inline paddle/phi/kernels/fusion/cutlass/cutlass_kernels/cutlass_heuristic.h]: sm86这个判断可能会修改到wint8, 加一下WINT4的per group限制吧

**Comment by wwbitejotunn** [comment]
可能还要注意下cutlass gemm编译后的目标文件大小，不能太大了

File: paddle/phi/kernels/fusion/cutlass/cutlass_kernels/cutlass_heuristic.h (hunk 1 above)
Comment (freeliuzc) [reply paddle/phi/kernels/fusion/cutlass/cutlass_kernels/cutlass_heuristic.h]: 已修改

**Comment by wwbitejotunn** [comment]
LGTM for mixgemm kernel

**Review by wwbitejotunn — APPROVED** [review]

**Review by qili93 — DISMISSED** [review]
LGTM for unittest.skipIf

**Review by qili93 — APPROVED** [review]
LGTM for unitskip

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by wwbitejotunn — APPROVED** [review]
LGTM for mix gemm kernel

**Pull request merged**